eng/validation/remove-oldbranches.ps1

```diff
@@ -0,0 +1,98 @@
+$global:azdoOrg = $azdoOrg
+$global:azdoProject = $azdoProject
+$global:azdoToken = $azdoToken
+$global:githubUser = $githubUser
+$global:githubPAT = $githubPAT
+$global:githubOrg = $githubOrg
+$global:githubRepoName = $githubRepoName
+$global:barToken = $barToken
+$global:azdoRepoName = if (-not $azdoRepoName) { "" } else { $azdoRepoName }
+
+# Get a temporary directory for a test root. Use the agent work folder if running under azdo, use the temp path if not.
+$testRootBase = if ($env:AGENT_WORKFOLDER) { $env:AGENT_WORKFOLDER } else { $([System.IO.Path]::GetTempPath()) }
+$testRoot = Join-Path -Path $testRootBase -ChildPath $([System.IO.Path]::GetRandomFileName())
+New-Item -Path $testRoot -ItemType Directory | Out-Null
+
+function GitHub-Clone($repoName) 
+{
+    & git clone $global:githubUri $(Get-Repo-Location $repoName)
+    Push-Location -Path $(Get-Repo-Location $repoName)
+    & git config user.email "${global:githubUser}@test.com"
+    & git config user.name $global:githubUser
+    Pop-Location
+}
+
+function Get-Repo-Location($repoName){ "$testRoot\$repoName" }
+
+function Git-Command($repoName) {
+    Push-Location -Path $(Get-Repo-Location($repoName))
+    try {
+        $gitParams = $args
+        if ($gitParams.GetType().Name -ne "Object[]") {
+            $gitParams = $gitParams.ToString().Split(" ")
+        }
+        Write-Host "Running 'git $gitParams' from $(Get-Location)"
+        $commandOutput = & git @gitParams; if ($LASTEXITCODE -ne 0) { throw "Git exited with exit code: $LASTEXITCODE" } else { $commandOutput }
+        $commandOutput
+    }
+    finally {
+        Pop-Location
+    }
+}
+
+## Global Variables
+$global:githubUri = "https://${global:githubUser}:${global:githubPAT}@github.com/${global:githubOrg}/${global:githubRepoName}"
+$global:azdoUri = "https://${global:githubUser}:${global:azdoToken}@dev.azure.com/${global:azdoOrg}/${global:azdoProject}/_git/${global:azdoRepoName}"
+$global:remoteName = ($global:azdoOrg + "-" + $global:azdoRepoName)
+$global:branchNamePrefix = "refs/heads/dev/" + $global:githubUser + "/arcade-"
+$global:darcAzDORepoName = "https://dev.azure.com/${global:azdoOrg}/${global:azdoProject}/_git/${global:azdoRepoName}"
+$global:lastBranch = $global:branchNamePrefix + $global:arcadeSdkVersion
+
+Write-Host "Cloning '${global:githubRepoName} from GitHub"
+GitHub-Clone $global:githubRepoName
+
+Write-Host "Add remote to Azure DevOps"
+Git-Command $global:githubRepoName remote add $remoteName $global:azdoUri
+
+$remoteBranches = @()
+Write-Host "Getting remote branches for '${global:githubRepoName}' on Azure DevOps"
+$remoteBranches = Git-Command $global:githubRepoName ls-remote --heads $global:azdoUri
+
+foreach($remoteBranch in $remoteBranches)
+{
+    ## Delete all of the old branches except for any that match the current version of Arcade being validated
+    if(($remoteBranch -match ".*($global:branchNamePrefix).*") -and -not ($remoteBranch -match ".*($global:lastBranch).*"))
```

Comment:
@missymessa - I'm quite concerned about what an error calling this script (or in the script.. RegExp!) would do. I don't see anything wrong but the potential effects if something goes wrong is pretty big. Can you make it safer to use this script?

Reply:
I could use `-like` instead of `-match` and get the same results. Do you think that would be better? What other things should I do here to make the script safer?

Reply:
I think I was more raising a concern than calling for something specific to be implemented.

Reply:
@riarenas pointed out that AzDO has an API that allows you to filter on branch names, so I'll give that a try to see if it works for this script rather than parsing results from git commands: https://docs.microsoft.com/en-us/rest/api/azure/devops/git/refs/list?view=azure-devops-rest-5.1#refs-heads-that-contain-a-word

Reply:
Refactored this script to make use of AzDO's API to look up branches and delete them. Much nicer code than all the git commands.